File: src/main/java/br/com/rotaractsorocabaleste/rotasales/sale/SaleService.java
package br.com.rotaractsorocabaleste.rotasales.sale;

import org.springframework.data.domain.Page;

import java.io.ByteArrayOutputStream;
import java.util.UUID;

public interface SaleService {

    Page<GetSalesVO> findMySales(final UUID event,
                                 final String buyer,
                                 final Boolean donation,
                                 final int page,
                                 final int limit);

    SaleDTO create(final SaleDTO saleDTO);

    void deleteById(final UUID saleId);

    SaleDTO findDTOById(final UUID saleId);

    void updateSale(final SaleDTO saleDTO);

    void changeSaleStatus(final ChangeSaleStatusDTO changeSaleStatusDTO);

    byte[] getSaleReceiptInPdfStream(final UUID saleId);
}

File: src/main/kotlin/br/com/rotaractsorocabaleste/rotasales/core/service/SaleService.kt
package br.com.rotaractsorocabaleste.rotasales.core.service

import br.com.rotaractsorocabaleste.rotasales.core.entity.Sale
import br.com.rotaractsorocabaleste.rotasales.core.vo.PatchSaleStatusRequestVO
import java.util.*

interface SaleService {
    fun create(sale: Sale)
    fun update(sale: Sale)
    fun delete(saleId: UUID)
    fun getLoggedInUserSales(): List<Sale>
    fun patchSaleStatus(patchSaleStatusRequestVO: PatchSaleStatusRequestVO)
    fun findById(saleId: UUID): Optional<Sale>
}